packages/gestalt/src/Button.js

```diff
@@ -84,22 +86,34 @@ type unionProps = ButtonType | SubmitButtonType | LinkButtonType;
 
 type unionRefs = HTMLButtonElement | HTMLAnchorElement;
 
-function IconEnd({
+function InternalButtonContent({
+  target,
   text,
   textColor,
   icon,
   size,
 }: {|
+  target?: null | 'self' | 'blank',
   text: Node,
   textColor: IconColor,
-  icon: $Keys<typeof icons>,
+  icon?: $Keys<typeof icons>,
   size: string,
 |}): Node {
   return (
-    <Flex alignItems="center" gap={{ row: 2, column: 0 }} justifyContent="center">
-      {text}
-      <Icon accessibilityLabel="" color={textColor} icon={icon} size={SIZE_NAME_TO_PIXEL[size]} />
-    </Flex>
+    <Fragment>
+      <Flex alignItems="center" gap={{ row: 2, column: 0 }} justifyContent="center">
+        {text}
+        {icon ? (
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={icon}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        ) : null}
+      </Flex>
+      {target === 'blank' ? <NewTabAccessibilityLabel target={target} /> : null}
```

Comment:
isn't it a bit redundant to check for target blank both here and within NewTabAccessibilityLabel?

Reply:
@ayeshakmaz It seems so! But no, this is expected. Flex doesn't know if the component is returning null so it adds gap spacing. I wanted to reuse NewTabAccessibilityLabel. The other components don't have the Flex wrapper, but button does, so I choose the redundancy so I could reuse NewTabAccessibilityLabel.

Reply:
Good catch. Initially it was all inside the Flex. Flex doesn't know if the component is returning null so it adds gap spacing if we don't keep the condition externally. I wanted to reuse NewTabAccessibilityLabel. The other components don't have the Flex wrapper, but button does, so I choose the redundancy so I could reuse NewTabAccessibilityLabel.

But then I moved it out, so then it's not needed it anymore.

Reply:
nice!